cantabular/metadata.go

```diff
@@ -0,0 +1,259 @@
+					} `graphql:"Statistical_Unit" json:"statistical_unit"`
+
+					Topic struct {
+						TopicMnemonic    graphql.String `graphql:"Topic_Mnemonic" json:"topic_mnemonic"`
+						TopicDescription graphql.String `graphql:"Topic_Description" json:"topic_description"`
+						TopicTitle       graphql.String `graphql:"Topic_Title" json:"topic_title"`
+					} `graphql:"Topic" json:"topic"`
+
+					VariableType struct {
+						VariableTypeCode        graphql.String `graphql:"Variable_Type_Code" json:"variable_type_code"`
+						VariableTypeDescription graphql.String `graphql:"Variable_Type_Description" json:"variable_type_description"`
+					} `graphql:"Variable_Type" json:"variable_type"`
+				} `graphql:"ONS_Variable" json:"ONS_Variable"`
+
+				Topics []struct {
+					TopicMnemonic    graphql.String `graphql:"Topic_Mnemonic" json:"topic_mnemonic"`
+					TopicDescription graphql.String `graphql:"Topic_Description" json:"topic_description"`
+					TopicTitle       graphql.String `graphql:"Topic_Title" json:"topic_title"`
+				} `graphql:"Topics" json:"topics"`
+			} `json:"meta"`
+			CatLabels *graphql.String `graphql:"catLabels" json:"catLabels,omitempty"`
+		} `graphql:"vars(names: $vars)" json:"vars"`
+	} `graphql:"dataset(name: $ds, lang: $lang)" json:"dataset"`
+}
+
+// params for GraphQL reqs
+type MetadataTableQueryRequest struct {
+	Lang      string   `json:"lang"`
+	Variables []string `json:"variables"`
+}
+
+type MetadataDatasetQueryRequest struct {
+	Dataset   string   `json:"dataset"`
+	Lang      string   `json:"lang"`
+	Variables []string `json:"variables"`
+}
+
+// MetadataTableQuery takes variables (typically a ONS dataset id) and language
+// params and returns a struct representing the metadata server response for a
+// table query which typically contain variables (dimensions) and Cantabular
+// dataset id
+
+func (c *Client) MetadataTableQuery(ctx context.Context, req MetadataTableQueryRequest) (*MetadataTableQuery, error) {
+	if c.gqlClient == nil {
+		return nil, dperrors.New(
+			errors.New("cantabular metadata client not configured"),
+			http.StatusServiceUnavailable,
+			nil,
+		)
+	}
+
+	logData := log.Data{
+		"url":     fmt.Sprintf("%s/graphql", c.extApiHost),
+		"request": req,
+	}
+
+	var datasetIDs []graphql.String
+	for _, v := range req.Variables {
+		datasetIDs = append(datasetIDs, graphql.String(v))
+	}
+
+	vars := map[string]interface{}{
+		"vars": datasetIDs,
+		"lang": graphql.String(req.Lang),
+	}
+
+	var fq MetadataTableQuery
+	if err := c.gqlClient.Query(ctx, &fq, vars); err != nil {
+		return nil, dperrors.New(
+			fmt.Errorf("failed to make GraphQL query: %w", err),
+			http.StatusInternalServerError,
+			logData,
+		)
+	}
+
+	return &fq, nil
+}
+
+// MetadataDatasetQuery takes cantabular dataset id, language and variables
+// (dimensions) params and returns a struct representing the metadata server
+// response for a dataset query
+
```

Comment:
Same, please remove this line

Reply:
Should be done - tested to display under `godoc -http=localhost:6060 -index`